daemon/lokinet-vpn.cpp

```diff
@@ -149,47 +161,37 @@ main(int argc, char* argv[])
     return 1;
   }
 
-  std::vector<std::string> firstHops;
-  std::string ifname;
-
-  const auto maybe_status = LMQ_Request(lmq, connID, "llarp.status");
-  if (not maybe_status.has_value())
+  if (printStatus)
   {
-    std::cout << "call to llarp.status failed" << std::endl;
-    return 1;
-  }
+    const auto maybe_status = LMQ_Request(lmq, connID, "llarp.status");
+    if (not maybe_status.has_value())
+    {
+      std::cout << "call to llarp.status failed" << std::endl;
+      return 1;
+    }
 
-  try
-  {
-    // extract first hops
-    const auto& links = maybe_status->at("result")["links"]["outbound"];
-    for (const auto& link : links)
+    try
     {
-      const auto& sessions = link["sessions"]["established"];
-      for (const auto& session : sessions)
+      const auto& ep = maybe_status->at("result").at("services").at(endpoint);
+      const auto exitMap = ep.at("exitMap");
+      if (exitMap.empty())
       {
-        std::string addr = session["remoteAddr"];
-        const auto pos = addr.find(":");
-        firstHops.push_back(addr.substr(0, pos));
+        std::cout << "no exits" << std::endl;
+      }
+      else
+      {
+        for (const auto& [range, exit] : exitMap.items())
+        {
+          std::cout << range << " via " << exit.get<std::string>() << std::endl;
+        }
       }
     }
-    // get interface name
-#ifdef _WIN32
-    // strip off the "::ffff."
-    ifname = maybe_status->at("result")["services"][endpoint]["ifaddr"];
-    const auto pos = ifname.find("/");
-    if (pos != std::string::npos)
+    catch (std::exception& ex)
     {
-      ifname = ifname.substr(0, pos);
+      std::cout << "failed to parse result: " << ex.what() << std::endl;
```

Comment:
It's not a big deal, but could go with a more detailed message here.  Like above you have "call to llarp.status failed", I think adding that context to this error message is good.